cmd/registry-experimental/cmd/vocabulary/similarity.go

```diff
@@ -27,101 +27,107 @@ import (
-			if err != nil {
-				return err
-			}
-			counts := make(map[string]int32)
-			total := 0
-
-			for _, artifactName := range []names.Artifact{
-				artifactName1, artifactName2,
-			} {
-				err = visitor.GetArtifact(ctx, client, artifactName, true,
-					func(ctx context.Context, artifact *rpc.Artifact) error {
-						messageType, err := mime.MessageTypeForMimeType(artifact.GetMimeType())
+			vocabularyArtifacts := make([]*rpc.Artifact, 0)
+			for _, p := range args {
+				pattern := c.FQName(p)
+				name, err := names.ParseArtifact(pattern)
+				if err != nil {
+					return err
+				}
+				if err = visitor.ListArtifacts(ctx, client,
+					name,
+					filter,
+					true, func(ctx context.Context, message *rpc.Artifact) error {
+						messageType, err := mime.MessageTypeForMimeType(message.GetMimeType())
 						if err != nil || messageType != "gnostic.metrics.Vocabulary" {
-							log.Debugf(ctx, "Skipping, not a vocabulary: %s", artifact.Name)
-							return nil
-						}
-						vocab := &metrics.Vocabulary{}
-						if err := proto.Unmarshal(artifact.GetContents(), vocab); err != nil {
-							log.FromContext(ctx).WithError(err).Debug("Failed to unmarshal contents")
+							log.Debugf(ctx, "Skipping, not a vocabulary: %s", message.Name)
 							return nil
 						}
+						vocabularyArtifacts = append(vocabularyArtifacts, message)
+						return nil
+					}); err != nil {
+					return err
+				}
+			}
+
+			fmt.Fprintf(cmd.OutOrStdout(), ",")
+			for i := 0; i < len(vocabularyArtifacts); i++ {
+				fmt.Fprintf(cmd.OutOrStdout(), "%s,", shortName(vocabularyArtifacts[i].Name))
+			}
+			fmt.Fprintf(cmd.OutOrStdout(), "\n")
+			for i := 0; i < len(vocabularyArtifacts); i++ {
+				vi := &metrics.Vocabulary{}
+				if err := proto.Unmarshal(vocabularyArtifacts[i].GetContents(), vi); err != nil {
+					log.FromContext(ctx).WithError(err).Debug("Failed to unmarshal contents")
+					return nil
+				}
+				fmt.Fprintf(cmd.OutOrStdout(), "%s,", shortName(vocabularyArtifacts[i].Name))
+				for j := 0; j < i; j++ {
+					fmt.Fprintf(cmd.OutOrStdout(), "%1.3f,", 0.0)
+				}
+				for j := i; j < len(vocabularyArtifacts); j++ {
+					vj := &metrics.Vocabulary{}
+					if err := proto.Unmarshal(vocabularyArtifacts[j].GetContents(), vj); err != nil {
+						log.FromContext(ctx).WithError(err).Debug("Failed to unmarshal contents")
+						return nil
+					}
+					counts := make(map[string]int32)
+					total := 0
+					for _, v := range []*metrics.Vocabulary{vi, vj} {
 						for _, wordlist := range [][]*metrics.WordCount{
-							vocab.Operations,
-							vocab.Schemas,
-							vocab.Parameters,
-							vocab.Properties,
+							v.Operations,
+							v.Schemas,
+							v.Parameters,
+							v.Properties,
 						} {
 							total += len(wordlist)
 							for _, pair := range wordlist {
 								counts[pair.Word] += pair.Count
 							}
 						}
-						return nil
-					})
-				if err != nil {
-					return err
+					}
+					// a = words unique to v1
```

Comment:
How are these comments helpful? I only found them distracting. If anything is unclear, I think we would be better off improving the variable names or commenting on the code itself instead of a parallel line of algorithmic comments.

Reply:
I was proving to myself that this was correct... if you don't think that's needed, I can take it out.